File: WebApi/api/api/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace api.Models
{
    public class Product
    {
        [Key]
        public long id { get; set; }

        [StringLength(22)]
        public string prod_code { get; set; }

        [StringLength(50)]
        public string prod_tname { get; set; }

        [StringLength(50)]
        public string prod_ename { get; set; }

        [StringLength(4)]
        public string uom_code { get; set; }

        [StringLength(14)]
        public string bar_code { get; set; }

        [StringLength(8)]
        public string entity { get; set; }

        [StringLength(10)]
        public string pdgrp_code { get; set; }

        [StringLength(10)]
        public string pdbrnd_code { get; set; }

        [StringLength(10)]
        public string pdtype_code { get; set; }

        [StringLength(10)]
        public string pddsgn_code { get; set; }

        [StringLength(10)]
        public string pdsize_code { get; set; }

        [StringLength(10)]
        public string pdcolor_code { get; set; }

        [StringLength(10)]
        public string  pdmisc_code { get; set; } = null;
       

        [StringLength(10)]
        public string pdmodel_code { get; set; } = null;

        [StringLength(100)]
        public string pdgrp_desc { get; set; } = null;

        [StringLength(100)]
        public string pdbrnd_desc { get; set; } = null;

        [StringLength(100)]
        public string pdtype_desc { get; set; } = null;

        [StringLength(100)]
        public string pddsgn_desc { get; set; } = null;

        [StringLength(100)]
        public string pdsize_desc { get; set; } = null;

        [StringLength(100)]
        public string pdcolor_desc { get; set; } = null;

        [StringLength(100)]
        public string pdmisc_desc { get; set; } = null;

        [StringLength(100)]
        public string pdmodel_desc { get; set; } = null;

        public decimal unit_price { get; set; }

        [StringLength(1)]
        public string prod_status { get; set; }

        [StringLength(15)]
        public string created_by { get; set; }

        public DateTime created_at { get; set; }
        [StringLength(15)]
        public string updated_by { get; set; }

        public DateTime updated_at { get; set; }
    }
}

File: web/src/app/_model/product.ts
import { AppSetting } from "../_constants/app-setting";

export class ProductSearchView {
    public pageIndex: number = 1;
    public itemPerPage: number = AppSetting.itemPerPage;
    public prod_code: string = "";
    public bar_code: string = "";
    public prod_tname: string = "";
    public pdbrnd_code: string = "";
    public pddsgn_code: string = "";
    public pdtype_code: string = "";
    public pdcolor_code: string = "";
    public pdsize_code: string = "";
    public uom_code: string = "";
    public status: string = "A";
    
    // public productModelText: string = "";
    // public productColorText: string = "";
    // public productSizeText: string = "";
    // public productDesignText: string = "";
    // public isSaleBed: boolean = true;
}

export class ProductView {
    public id: number = undefined;
    public prod_code: string = "";
    public bar_code: string = "";
    public prod_tname: string = "";
    public pdtype_code: string = "";
    public pdbrnd_code: string = "";
    public pddsgn_code: string = "";
    public pdcolor_code: string = "";
    public pdsize_code: string = "";
    public uom_code: string = "";
    public status: string = "";
    public statusText: string = "";
    public pdtype_desc: string = "";
    public pdbrnd_desc: string = "";
    public pddsgn_desc: string = "";
    public pdcolor_desc: string = "";
    public pdsize_desc: string = "";
    public unit_price: number = 0;
       
}

export class ProductSyncSearchView
{
    public pddsgn_code : string = "";
}
